lib/matplotlib/axes/_base.py

```diff
@@ -438,8 +438,12 @@ def __init__(self, fig, rect,
         self._sharex = sharex
         self._sharey = sharey
         if sharex is not None:
+            if type(sharex) is bool:
```

Comment:
is there a reason we do not do
~~~
if not isinstance(sharex, _AxesBase):
~~~
(or `Axes`). If we check anyway, we can check against the expected type and not only exclude one non-expected type.